components/mpas-ocean/src/shared/mpas_ocn_vel_forcing_topographic_wave_drag.F

```diff
@@ -162,13 +181,100 @@ subroutine ocn_vel_forcing_topographic_wave_drag_init(err)!{{{
+! Nbar = 1300 N0 (1 - exp(-H/1300))/H
+              do iEdge = 1, nEdgesAll
+                  kmax =  maxLevelEdgeTop(iEdge)
+                  kmin =  minLevelEdgeBot(iEdge)
+                  topographicWaveDrag(iEdge) = 0.0_RKIND
+                  if (kmax > 0) then
+                      k = 1 ! Note: Only works for single layer case
+                      topoDragTurnOff = (TANH((layerThickEdgeMean(k,iEdge)-topoDragCutoffDepth)/topoDragCutoffWidth)+1)/2
+                      topographicWaveDrag(iEdge) = topographicWaveDragCoeff*topoDragTurnOff*50.0_RKIND  * &
+                                (bed_slope_edges(iEdge)**2)                                             * &
+                                (5.24e-3_RKIND*1300.0_RKIND                                             * &
+                                    (1 - exp(-layerThickEdgeMean(k,iEdge)/1300.0_RKIND))                / &
+                                        layerThickEdgeMean(k,iEdge))                                    * &
+                                (5.24e-3_RKIND*exp(-layerThickEdgeMean(k,iEdge)/1300.0_RKIND))          / &
+                                (8 * omegaM2 * pii**2)
+                  endif
+              enddo
+          else if (config_topographic_wave_drag_scheme.EQ."LGF") then
+              tensorScheme = .true.
+              call mpas_log_write("")
+              call mpas_log_write(" Topographic wave drag scheme is: Local Generation Formula")
+              call mpas_log_write("")
+              do iEdge = 1, nEdgesAll
+                  kmax =  maxLevelEdgeTop(iEdge)
+                  kmin =  minLevelEdgeBot(iEdge)
+                  topographicWaveDrag(iEdge) = 0.0_RKIND
+                  if (kmax > 0) then
+                      k = 1
+                      topoDragTurnOff = (TANH((layerThickEdgeMean(k,iEdge)-topoDragCutoffDepth)/topoDragCutoffWidth)+1)/2
+                      topographicWaveDrag = topographicWaveDragCoeff  &
+                                            * topoDragTurnOff         &
+                                            * (sqrt((topo_buoyancy_N1B(iEdge)**2 - omegaM2**2) * &
+                                                    (topo_buoyancy_N1V(iEdge)**2 - omegaM2**2))) &
+                                            / (4*pii*omegaM2)
+                      normalCoeff = (lonGradEdge(iEdge)**2) * &
+                                    (cos(angleEdge(iEdge))**2) + (latGradEdge(iEdge)**2)*(sin(angleEdge(iEdge))**2) - &
+                                        2*latGradEdge(iEdge)*lonGradEdge(iEdge)*sin(angleEdge(iEdge))*(cos(angleEdge(iEdge)))
+                      tangentialCoeff = sin(angleEdge(iEdge)) * &
+                                        cos(angleEdge(iEdge)) * &
+                                        (lonGradEdge(iEdge)**2 - latGradEdge(iEdge)**2) + &
+                                        latGradEdge(iEdge)*lonGradEdge(iEdge)*(cos(angleEdge(iEdge))**2 - sin(angleEdge(iEdge))**2)
```

Comment:
These variables should be indexed, i.e., `topographicWaveDrag(iEdge) = `, `normalCoeff(iEdge) =`, `tangentialCoeff(iEdge) = `

Reply:
I'm not sure what fortran will do in this case, but I feel like testing for `LGF` should be revisited...

Reply:
I'm pretty sure it'll set the entire array to a single value, so this definitely needs to be retested.